core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -675,9 +676,8 @@ private[zio] final class FiberContext[E, A](
     val oldState = state.get
 
     oldState match {
-      case Executing(FiberStatus.Suspended, observers) =>
-        if (!state.compareAndSet(oldState, Executing(FiberStatus.Running, observers)))
-          exitAsync()
+      case Executing(_: FiberStatus.Suspended, observers, interrupt) =>
```

Comment:
How about `FiberStatus.Suspended(_)`, to avoid the appearance of type casing?

Reply:
I'm not sure if that would result in a call to .unapply, avoiding that is the only reason I used a type match here, will check...

Reply:
Ah, performance is always a reason to break rules in ZIO. 😆

Reply:
Hmm, I'm wrong! Scala is smart enough to avoid `unapply` not only here, but everywhere in `FiberContext` there are no unapply calls:
```java
if (var6 instanceof FiberContext$FiberStatus$Suspended) {
   if (!this.state.compareAndSet(oldState, new FiberContext$FiberState$Executing(FiberContext$FiberStatus$Running$.MODULE$, observers, interrupt))) {
      continue;
   }

   var2 = true;
   break;
}
```